### ISSUE TITLE
No grins after restore, strange accounts

### ISSUE THREAD

**Issue by phooton** [mimblewimble/grin#2296]
Happened when mining and sending grins via faucet. Faucet transactions started to fail, saw strangely grouped recent three coinbases and I restored without thinking too much about it.

```gringod@gringod:~/grin/target/release$ ./grin --floonet  wallet  restore
20190106 00:42:28.635 INFO grin_util::logger - log4rs is initialized, file level: Error, stdout level: Debug, min. level: Debug
20190106 00:42:28.635 INFO grin - Using wallet configuration file at /home/gringod/.grin/floo/grin-wallet.toml
20190106 00:42:28.635 INFO grin - This is Grin version 0.5.0 (git v0.5.0-7-g5e86cc8), built for x86_64-unknown-linux-gnu by rustc 1.33.0-nightly (09d6ab90e 2018-12-20).
20190106 00:42:28.635 DEBUG grin - Built with profile "release", features "".
Password:
20190106 00:42:30.523 DEBUG grin_wallet::types - Using wallet seed file at: /home/gringod/.grin/floo/wallet_data/wallet.seed
20190106 00:42:30.548 INFO grin_wallet - Using LMDB Backend for wallet
20190106 00:42:30.548 DEBUG grin_wallet::types - Using wallet seed file at: /home/gringod/.grin/floo/wallet_data/wallet.seed
20190106 00:42:30.572 INFO grin_wallet::libwallet::internal::restore - Starting restore.
20190106 00:42:30.830 INFO grin_wallet::libwallet::internal::restore - Retrieved 1000 outputs, up to index 35709. (Highest index: 3476)
20190106 00:42:30.831 INFO grin_wallet::libwallet::internal::restore - Scanning 1000 outputs in the current Grin utxo set
20190106 00:42:31.419 INFO grin_wallet::libwallet::internal::restore - Retrieved 1000 outputs, up to index 35709. (Highest index: 7003)
20190106 00:42:31.420 INFO grin_wallet::libwallet::internal::restore - Scanning 1000 outputs in the current Grin utxo set
20190106 00:42:31.974 INFO grin_wallet::libwallet::internal::restore - Retrieved 1000 outputs, up to index 35709. (Highest index: 10840)
20190106 00:42:31.975 INFO grin_wallet::libwallet::internal::restore - Scanning 1000 outputs in the current Grin utxo set
20190106 00:42:32.529 INFO grin_wallet::libwallet::internal::restore - Retrieved 1000 outputs, up to index 35709. (Highest index: 18681)
20190106 00:42:32.529 INFO grin_wallet::libwallet::internal::restore - Scanning 1000 outputs in the current Grin utxo set
20190106 00:42:33.159 INFO grin_wallet::libwallet::internal::restore - Retrieved 1000 outputs, up to index 35709. (Highest index: 29682)
20190106 00:42:33.160 INFO grin_wallet::libwallet::internal::restore - Scanning 1000 outputs in the current Grin utxo set
20190106 00:42:33.733 INFO grin_wallet::libwallet::internal::restore - Retrieved 1000 outputs, up to index 35712. (Highest index: 32040)
20190106 00:42:33.733 INFO grin_wallet::libwallet::internal::restore - Scanning 1000 outputs in the current Grin utxo set
20190106 00:42:34.276 INFO grin_wallet::libwallet::internal::restore - Retrieved 1000 outputs, up to index 35712. (Highest index: 33399)
20190106 00:42:34.277 INFO grin_wallet::libwallet::internal::restore - Scanning 1000 outputs in the current Grin utxo set
20190106 00:42:34.833 INFO grin_wallet::libwallet::internal::restore - Retrieved 1000 outputs, up to index 35712. (Highest index: 34784)
20190106 00:42:34.833 INFO grin_wallet::libwallet::internal::restore - Scanning 1000 outputs in the current Grin utxo set
20190106 00:42:34.971 INFO grin_wallet::libwallet::internal::restore - Output found: Commitment(081e80cfddc801f94d330816b7b24c2ff972a52ee9ac4cd7e9f9c0edcfc785b168), amount: 60000000000, parent_key_id: Identifier(03c0820070767f00000000006100000000)
20190106 00:42:35.312 INFO grin_wallet::libwallet::internal::restore - Retrieved 724 outputs, up to index 35712. (Highest index: 35712)
20190106 00:42:35.312 INFO grin_wallet::libwallet::internal::restore - Scanning 724 outputs in the current Grin utxo set
20190106 00:42:35.361 INFO grin_wallet::libwallet::internal::restore - Output found: Commitment(092ac819f313b9474cf9a5b08ff703be91062ed34de714f29b811a5465f9d31342), amount: 2417261000000, parent_key_id: Identifier(03000100000000000000a4da44e5550000)
20190106 00:42:35.467 INFO grin_wallet::libwallet::internal::restore - Output found: Commitment(08c65d466cfcba8e13d2bd3edb3bdbaa04b8267a9c839580dc1cad1041f884a333), amount: 60007000000, parent_key_id: Identifier(0360800098727f00000000006a00000000)
20190106 00:42:35.472 INFO grin_wallet::libwallet::internal::restore - Output found: Commitment(097e0dfca3657726876975930f70bf544d5be625683b669c191e61c8f37fd44591), amount: 60007000000, parent_key_id: Identifier(0350810098727f00000000006b00000000)
20190106 00:42:35.485 INFO grin_wallet::libwallet::internal::restore - Output found: Commitment(085d5e365fc41a899c8eff3ec4baaf1b7a3db8cb7b122bed1adcc98cb6d575a6f1), amount: 60000000000, parent_key_id: Identifier(03a0760080ef7e00000000006c00000000)
20190106 00:42:35.486 INFO grin_wallet::libwallet::internal::restore - Output found: Commitment(090404ffceed5d5a9c891ad58240b715eb90bbf5cc4b3fdc42de9540f602b79e75), amount: 60000000000, parent_key_id: Identifier(03e0210080ef7e00000000006d00000000)
20190106 00:42:35.534 INFO grin_wallet::libwallet::internal::restore - Identified 6 wallet_outputs as belonging to this wallet
20190106 00:42:35.587 INFO grin_wallet::command - Wallet restore complete
Command 'restore' completed successfully
```
```
____ Wallet Summary Info - Account 'default' as of height 11594 ____

 Total                            | 0.000000000
 Awaiting Confirmation (< 10)     | 0.000000000
 Locked by previous transaction   | 0.000000000
 -------------------------------- | -------------
 Currently Spendable              | 0.000000000
```
```
gringod@gringod:~/grin/target/release$ grin --floonet wallet account
20190106 00:46:59.110 INFO grin_util::logger - log4rs is initialized, file level: Error, stdout level: Debug, min. level: Debug
20190106 00:46:59.110 INFO grin - Using wallet configuration file at /home/gringod/.grin/floo/grin-wallet.toml
20190106 00:46:59.110 INFO grin - This is Grin version 0.5.0 (git v0.5.0-7-g5e86cc8), built for x86_64-unknown-linux-gnu by rustc 1.33.0-nightly (09d6ab90e 2018-12-20).
20190106 00:46:59.110 DEBUG grin - Built with profile "release", features "".
Password:
20190106 00:47:01.756 DEBUG grin_wallet::types - Using wallet seed file at: /home/gringod/.grin/floo/wallet_data/wallet.seed
20190106 00:47:01.766 INFO grin_wallet - Using LMDB Backend for wallet

____ Wallet Accounts ____

 Name      | Parent BIP-32 Derivation Path
-----------+-------------------------------
 account_1 | m/65536/0
 account_2 | m/1350631576/1920925696
 account_3 | m/3760259200/4018012160
 account_4 | m/3229745264/1988034560
 account_5 | m/2692087936/4018012160
 account_6 | m/1619001496/1920925696
 default   | m/0/0

Command 'account' completed successfully

```
Starting with fresh node has no effect. Let me know if I need to find something in the logs or send my seed file to someone.

**Comment by jaspervdm** [comment]
Probably related: https://github.com/mimblewimble/rust-secp256k1-zkp/issues/41

**Comment by yeastplume** [comment]
@phooton Can you share either the recovery phrase or the seed file with us (can PM via Gitter if you like). We need to take a closer look a these outputs, firstly to determine if this 'corruption' is on the way in or out.

**Comment by yeastplume** [comment]
Should be addressed by #2304, thanks for reporting this
